### ISSUE TITLE
Swap Disk on PSX freezes

### ISSUE THREAD

**Issue by saljumerah** [jasarien/Provenance#728]
Hi,

so, I got the zip file with Final Fantasy 8 into the App. It shows all the disks as PSX games. However, the M3U file remains in the roms-folder. Is that correct?

I followed the PSX instructions. It was one zip file with all the 4 .bin and .iso files plus the 1 .m3u file.

Disk 1 runs more or less smooth. It‘s playable. When I try to swap the disk, the game just freezes though. I can reset or go back to the menu and restart this game or another one. But every time I try to swap, nothing happens.

I‘m using an iPhone 6S Plus.

Is that normal? Or am I missing something?

Thanks for your help! :)

**Comment by gh0sti** [comment]
I believe they are still working on this issue and is not fully completed yet, I would wait till v1.5 is released or 2.0 where disc swap will be fully featured into the app.

**Comment by sevdestruct** [comment]
No, this is still a very recent feature, and is in testing. As for the m3u file, it needs to be with the .bin/cue files in com.provenance.psx or it won't work. Nothing should remain in roms/ (which is changing soon to Imports/). There is work being done currently improve the importer, so this is a known issue.
Need to split apart this issues in this post…
.iso files, don't think those are supported, you'll need to convert them to .bin/cue or scrap them and redownload files mednafen (our psx core) supports .bin/.cue per disc.

**Comment by saljumerah** [comment]
Sorry, I meant .cue and not .iso. 

But thanks for your replies so far. Can I just move the m3u manually? Would that help somehow?

**Comment by sevdestruct** [comment]
@saljumerah: Yes, via Web Server UI you can, which is the current workaround until this is fixed…

**Comment by JoeMatt** [comment]
I haven't seen this in my multi-disc testing, and I don't want this to be a blocker to 1.5. 

Can check back in the future if this is an issue, multi-disc was a last minute add on I did out of the kindness of my heart anyway, if it's a little buggy so be it for now, it didn't exist at all for a long time.

**Comment by sevdestruct** [comment]
**In the mean while,** anyone running into issues with multi-disc games, please refer to the updated wiki for [Multi-file ROMs / Muilti-disc Games](https://github.com/jasarien/Provenance/wiki/Importing-ROMs#multi-file-roms)

**Comment by kaiser71** [comment]
How would I use the game shark or action replay?? You have to swap discs with those.

**Comment by sevdestruct** [comment]
Not sure I can answer that, being that those are cheatcode related, it's more likely you would simulate/replicate what those actually did instead of inserting the disc.
